include/higra/attribute/tree_attribute.hpp

```diff
@@ -196,18 +196,15 @@ namespace hg {
         hg_assert_node_weights(tree, altitudes);
         hg_assert_1d_array(altitudes);
 
-        tree.compute_children();
-        array_1d<bool> extrema = xt::zeros<bool>({num_vertices(tree)});
-        for (auto n: leaves_to_root_iterator(tree, leaves_it::exclude)) {
-            bool flag = true;
-            for (auto c: children_iterator(n, tree)) {
-                bool c_non_canonical = altitudes(c) == altitudes(n);
-                if (!(is_leaf(c, tree) || (c_non_canonical && extrema(c)))) {
-                    flag = false;
-                }
-                extrema(c) = extrema(c) && !c_non_canonical;
-            }
-            extrema(n) = flag;
+        array_1d<bool> extrema = xt::ones<bool>({num_vertices(tree)});
+
+        const auto num_leaves = (hg::index_t) hg::num_leaves(tree);
+        xt::noalias(xt::view(extrema, xt::range(0, num_leaves))) = xt::zeros<bool>({num_leaves});
```

Comment:
The allocation of an explicit 0 array is not needed here, we can use broadcasting :

```cpp
xt::view(extrema, xt::range(0, num_leaves)) = 0;
```